ledger-service/http-json/src/main/scala/com/digitalasset/http/HttpService.scala

```diff
@@ -261,13 +262,72 @@ object HttpService {
   }
 
   private[http] def httpsConnectionContext(config: TlsConfiguration): HttpsConnectionContext = {
-    import io.netty.buffer.ByteBufAllocator
-    val sslContext =
-      config.server
-        .getOrElse(
-          throw new IllegalArgumentException(s"$config could not be built as a server ssl context")
-        )
-    ConnectionContext.httpsServer(() => sslContext.newEngine(ByteBufAllocator.DEFAULT))
+    import java.security.SecureRandom
+    import javax.net.ssl.{SSLContext, KeyManagerFactory, TrustManagerFactory}
+
+    val keyStore = buildKeyStore(
+      config.certChainFile.get.toPath,
+      config.privateKeyFile.get.toPath,
+      config.trustCollectionFile.get.toPath,
+    )
+
+    val keyManagerFactory = KeyManagerFactory.getInstance("SunX509")
+    keyManagerFactory.init(keyStore, null)
+
+    val trustManagerFactory = TrustManagerFactory.getInstance("SunX509")
+    trustManagerFactory.init(keyStore)
+
+    val context = SSLContext.getInstance("TLS")
+    context.init(
+      keyManagerFactory.getKeyManagers,
+      trustManagerFactory.getTrustManagers,
+      new SecureRandom,
+    )
+
+    ConnectionContext.httpsServer(context)
+  }
+
+  def buildKeyStore(certFile: Path, privateKeyFile: Path, caCertFile: Path): KeyStore = {
+    import java.security.cert.CertificateFactory
+    import scala.util.Using
+
+    val alias = "key" // This can be anything as long as it's consistent.
+
+    val cf = CertificateFactory.getInstance("X.509")
+    val cert = Using.resource(Files.newInputStream(certFile)) { cf.generateCertificate(_) }
+    val caCert = Using.resource(Files.newInputStream(caCertFile)) { cf.generateCertificate(_) }
+    val privateKey = loadPrivateKey(privateKeyFile)
+
+    val keyStore = KeyStore.getInstance("PKCS12")
+    keyStore.load(null)
+    keyStore.setCertificateEntry(alias, cert)
+    keyStore.setCertificateEntry(alias, caCert)
+    keyStore.setKeyEntry(alias, privateKey, null, Array(cert, caCert))
+    keyStore
+  }
+
+  def loadPrivateKey(pkRsaPemFile: Path): Key = {
+    // TODO: Use a library to support other private key formats?
```

Comment:
I didn't want to introduce a new library dependency while other library update machinations are in flux, so just handle the existing example format for now.